api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -40,6 +55,36 @@
 	@Autowired
 	private ObservationCategoryMap categoryMap;
 	
+	@Autowired
+	@Getter(AccessLevel.PUBLIC)
+	@Setter(AccessLevel.PUBLIC)
+	@Qualifier("sessionFactory")
+	private SessionFactory sessionFactory;
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<String> getSearchResultUuids(@Nonnull SearchParameterMap theParams) {
+		if (!theParams.getParameters(FhirConstants.LASTN_SEARCH_HANDLER).isEmpty()) {
+			
+			Criteria criteria = sessionFactory.getCurrentSession().createCriteria(typeToken.getRawType());
+			
+			setupSearchParams(criteria, theParams);
+			
+			criteria.setProjection(Projections.projectionList().add(Projections.property("uuid"))
+			        .add(Projections.property("concept")).add(Projections.property("obsDatetime")));
+			
+			@SuppressWarnings("unchecked")
+			List<Object[]> results = criteria.list();
```

Comment:
Consider instead of just passing this `List<Object[]>` structure around using a typed holder class, e.g.,

```java
// as an inner class
@Value
private static final class LastnObservationResult {
	private String uuid;
	private Concept concept;
	private Date obsDateTime;

	LastnObservationResult(List<Object[]> values) {
		this.uuid = (String) values[0];
		this.concept = (Concept) values[1];
		this.obsDateTime = (Date) values[2];
	}
}
```

This will be easier to pass around and you won't have to do casting everywhere.